File: models/Order.js
const mongoose = require("mongoose");

const orderSchema = new mongoose.Schema(
  {
    // 👤 CUSTOMER ACCOUNT LINK (for login + order history)
    customerId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User"
    },

    // 👤 Customer info (snapshot)
    customerName: {
      type: String,
      required: true,
      trim: true
    },
    phone: {
      type: String,
      required: true
    },
    email: {
      type: String
    },
    address: {
      type: String,
      required: true
    },

    // 🛍️ Items
    items: [
      {
        name: String,
        price: Number,
        qty: Number,
        productId: {
          type: mongoose.Schema.Types.ObjectId,
          ref: "Product"
        }
      }
    ],

    // 💰 Amount
    total: {
      type: Number,
      required: true
    },

    // 💳 Payment
    paymentStatus: {
      type: String,
      enum: ["PENDING", "PAID", "FAILED"],
      default: "PENDING"
    },
    paymentId: String,
    razorpayOrderId: String,

    // 📦 Order lifecycle (admin)
    status: {
      type: String,
      enum: ["PENDING", "PROCESSING", "DELIVERED", "CANCELLED"],
      default: "PENDING"
    },

    // 🚚 DELIVERY TRACKING
    courierName: {
      type: String
    },

    trackingNumber: {
      type: String
    },

    trackingStatus: {
      type: String,
      enum: ["ORDER_PLACED", "SHIPPED", "OUT_FOR_DELIVERY", "DELIVERED"],
      default: "ORDER_PLACED"
    },

    statusHistory: [
      {
        status: String,
        time: {
          type: Date,
          default: Date.now
        }
      }
    ],

    // 🧾 Extra
    notes: String
  },
  {
    timestamps: true
  }
);

module.exports = mongoose.model("Order", orderSchema);

File: classy_crafth_app/lib/models/order.dart
class Order {
  final String id;
  final String customerName;
  final String phone;
  final String address;
  final double total;
  final List items;
  final String createdAt;

  Order({
    required this.id,
    required this.customerName,
    required this.phone,
    required this.address,
    required this.total,
    required this.items,
    required this.createdAt,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json["_id"] ?? "",
      customerName: json["customerName"] ?? "",
      phone: json["phone"] ?? "",
      address: json["address"] ?? "",
      total: double.parse(json["total"].toString()),
      items: json["items"] ?? [],
      createdAt: json["createdAt"] ?? "",
    );
  }
}
